jr-objects/src/main/java/com/fasterxml/jackson/jr/ob/impl/BeanReader.java

```diff
@@ -155,6 +155,20 @@ public Object read(JSONReader r, JsonParser p) throws IOException
                 return _constructors.create(p.getLongValue());
             case START_OBJECT:
                 {
+                    if (RecordsHelpers.isRecord(_valueType)) {
```

Comment:
To avoid doing this check on every read, could (and should) have a `boolean _isRecord` initialized in constructor.

Reply:
eb6521bef69dfe29b947d224ed8aa227c6af071f